library/entropy.c

```diff
@@ -466,6 +466,80 @@ static int entropy_dummy_source( void *data, unsigned char *output,
     return( 0 );
 }
 
+#if !defined(MBEDTLS_TEST_NULL_ENTROPY) && defined(MBEDTLS_ENTROPY_HARDWARE_ALT)
+
+static int entropy_self_test_bias_gather( unsigned char *buf, size_t buf_len )
+{
+    int ret = 0;
+    size_t entropy_len = 0;
+    size_t olen;
+    size_t attempts = buf_len;
+
+    while( attempts > 0 && entropy_len < buf_len )
+    {
+        if( ( ret = mbedtls_hardware_poll( NULL, buf + entropy_len,
+            buf_len - entropy_len, &olen ) ) != 0 )
+            return( ret );
+
+        entropy_len += olen;
+        attempts--;
+    }
+
+    if( entropy_len < buf_len )
+    {
+        ret = 1;
+    }
+
+    return( ret );
+}
+
+/*
+ * A quick test to ensure hat the entropy sources are functioning correctly
```

Comment:
Please remove 'quick' - it may not be a quick test and is dependent on how long it takes to read in the entropy - which can take a long time dependent on source.